src/app-bundles/maps-bundle.js

```diff
@@ -9,79 +9,103 @@ import ScaleBar from "ol/control/ScaleLine";
 import BasemapPicker from "../ol-controls/basemap-picker";
 
 const actions = {
-  MAPS_INITIALIZED: `MAPS_INITIALIZED`,
-  MAPS_SHUTDOWN: `MAPS_SHUTDOWN`,
+  MAPS_INITIALIZED: "MAPS_INITIALIZED",
+  MAPS_SHUTDOWN: "MAPS_SHUTDOWN",
+  MAPS_LOADED: "MAPS_LOADED",
 };
 
-export default {
-  name: "maps",
+export default (() => {
+  const internalState = {
+    map: null,
```

Comment:
I think part of the idea with this "generic" bundle is to let it handle more than one map. So I think we want the internal state use a key, so that it can track multiple maps. So rather than "map", I think this will be an empty object, and new maps will be added (and existing maps retrieved) based on a specified key. Thoughts?

Reply:
```
{
  "locationMap": {
    map: //map instance
    layer: //layer instance if we need it?
    overlay: //I think at some point we'll need to track the overlay 
             //so we can remove click listener when overlay is removed
    initializedMap: true,
    mapLoaded: true
  },
  "someOtherMap": {
    ...
  }
}
```

Reply:
I guess if we keep the non-OL-objects in the store state, the state would end up looking something like this, using the same map keys that are used in internalState (where the OL objects are kept):

```
{
  "locationMap": {
    initializedMap: true,
    mapLoaded: true,
    mapState: {
      // I just copied these from the @corpsmap-bundles/bundles/create-ol-map-bundle
      center: [-98.579, 39.828],
      zoom: 5,
      rotation: 0,
      minZoom: 0,
      maxZoom: 22,
      basemaps: basemaps, // I guess this comes from @corpsmap-bundles/config/basemaps.js
      activeIdx: 0,
    }
  },
  "insetMap": {
    ...
  }
}
```

Anyway, I am just throwing out ideas. Try whatever you think makes sense and we'll try to figure out what works best.